systems/controllers/constrained_lqr_controller.cc

```diff
@@ -107,37 +99,48 @@ ConstrainedLQRController::ConstrainedLQRController(
   // To compute the linearization, xdot (autodiff) is computed
   // Creating a combined autodiff vector and then extracting the individual
   // components to ensure proper gradient initialization.
-  VectorXd xul(num_states_ + num_efforts_ + num_forces_);
-  xul << x, u, lambda;
-  AutoDiffVecXd xul_autodiff = initializeAutoDiff(xul);
-  AutoDiffVecXd x_autodiff = xul_autodiff.head(num_states_);
-  AutoDiffVecXd u_autodiff = xul_autodiff.segment(num_states_, num_efforts_);
-  AutoDiffVecXd lambda_autodiff = xul_autodiff.tail(num_forces_);
 
-  // xdot
-  AutoDiffVecXd xdot_autodiff = contact_toolkit_->CalcTimeDerivatives(
-      x_autodiff, u_autodiff, lambda_autodiff);
+  VectorXd xu(plant_.num_positions() + plant_.num_velocities()
+      + plant_.num_actuators());
+  auto x = autoDiffToValueMatrix(plant_.GetPositionsAndVelocities(context));
+  auto u =
+      autoDiffToValueMatrix(plant_.get_actuation_input_port().Eval(context));
+  xu << x, u;
+  AutoDiffVecXd xu_ad = initializeAutoDiff(xu);
+
+  AutoDiffVecXd x_ad = xu_ad.head(plant_.num_positions()
+      + plant_.num_velocities());
+  AutoDiffVecXd u_ad = xu_ad.segment(plant_.num_positions()
+      + plant_.num_velocities(), plant_.num_actuators());
+
+  auto context_ad = multibody::createContext(plant_, x_ad, u_ad);
 
-  // Making sure that the derivative is zero
-  // DRAKE_DEMAND(autoDiffToValueMatrix(xdot_autodiff).isZero(1e-6));
+  AutoDiffVecXd xdot = evaluators_.CalcTimeDerivatives(*context_ad);
```

Comment:
It seems that the reason we pass the autodiff version of evaluator as a constructor's argument is because of this line and line 120. 
If we can have a method that creates `KinematicEvaluatorSet<Autodiff>` from `KinematicEvaluatorSet<double>`, then the user won't need to pass the autodiff one, and the constructor will probably be cleaner.

Reply:
I don't know if it's possible to convert Context<double> to Context<Autodiff>

Reply:
Agreed--but I'm not sure there's a simple way to do that (without having to use `new` and memory management). You'd essentially need to deep copy the `KinematicEvaluatorSet`, which would require copying the `KinematicEvaluator` objects--but the `Set` only stores these as pointers.